test/unit/mrd/test_srsnv_train_utils.py

```diff
@@ -0,0 +1,85 @@
+import os
+from collections import defaultdict
+from os.path import join as pjoin
+from test import get_resource_dir
+
+import pysam
+
+from ugvc.mrd.srsnv_training_utils import prepare_featuremap_for_model
+
+inputs_dir = get_resource_dir(__file__)
+
+
+def __count_variants(vcf_file):
+    counter = 0
+    for _ in pysam.VariantFile(vcf_file):
+        counter += 1
+    return counter
+
+
+def test_prepare_featuremap_for_model(tmpdir):
+    """Test that downsampling training-set works as expected"""
+
+    input_featuremap_vcf = pjoin(
+        inputs_dir,
+        "333_CRCs_39_LAv5and6.featuremap.single_substitutions.subsample.vcf.gz",
+    )
+    downsampled_training_featuremap_vcf, _, _ = prepare_featuremap_for_model(
+        workdir=tmpdir,
+        input_featuremap_vcf=input_featuremap_vcf,
+        train_set_size=12,
+        balanced_sampling_info_fields=None,
+        random_seed=0,
+    )
+
+    # Since we use random downsampling the train_set_size might differ slightly from expected
+    assert __count_variants(downsampled_training_featuremap_vcf) == 11
+
+
+def test_prepare_featuremap_for_model_with_motif_balancing(tmpdir):
+    """Test that downsampling training-set works as expected"""
+
+    input_featuremap_vcf = pjoin(
+        inputs_dir,
+        "333_LuNgs_08.annotated_featuremap.vcf.gz",
+    )
+    balanced_sampling_info_fields = ["trinuc_context", "is_forward"]
```

Comment:
do we still have trinuc_contextas a feature? I'm confused.

Reply:
Oh. There's an older file... do you insist I change it?